build/wrapper.cc

```diff
@@ -187,7 +192,10 @@ ObjectID put(const std::shared_ptr<RayObject> object,
 
     // Store our data in the object store
     ObjectID object_id;
-    RAY_CHECK_OK(worker.Put(*object, contained_object_ids, &object_id));
+    auto status = worker.Put(*object, contained_object_ids, &object_id);
+    if (!status.ok()) {
+        throw std::runtime_error(status.ToString());
+    }
     return object_id;
```

Comment:
So the actual [`CoreWorker::Put`](https://github.com/ray-project/ray/blob/a03efd9931128d387649dd48b0e4864b43d3bfb4/src/ray/core_worker/core_worker.h#L558-L560) returns just `Status` and passes back the `object_id` by reference. I'd try that approach here.

Reply:
It's definitely a little bit of scope creep but it should clean things up nicely